feat(js): Update scope docs for v8 (#9907)
getsentry/sentry-docs: docs/platforms/javascript/common/enriching-events/scopes/index.mdx

for you in the framework integrations and you don't need to think about them. However,
you should know what a scope is and how you can use it for your advantage.

> ## What's a Scope

Reviewer: ```suggestion
## What's a Scope?
```

The scope will hold useful information that should be sent along with the
event. For instance [contexts](../context/) or
[breadcrumbs](../breadcrumbs/) are stored on

> the scope. When a scope is forked, it inherits all data from the parent scope.

Reviewer: l: I'm a bit on the fence if we should publicly speak about "forking" a scope. It's one of these terms that not all devs will understand immediately. Though to be honest, just saying "cloning" is probably not descriptive enough and something like "create scopes" doesn't imply that we inherit data... (especially in the next paragraph). Not sure, feel free to leave it as-is; this is an expert section anyway

Author: yeah, I know what you mean, I was also not 100% sure there... 🤔

The default SDK integrations will fork scopes intelligently. For

> instance web framework integrations will fork scopes around your
> routes or controllers.

Reviewer: l: Maybe a bit closer to the truth, wdyt? 🤔 

```suggestion
instance web framework integrations will fork scopes around your
routes or request handlers.
```

## How Scopes Work

> Scopes are basically a stack of data that is attached to events. When an event is captured, the SDK will merge the data from the current scope into the event. This allows you to attach data to events that is relevant to the context in which the event was captured.

Reviewer 2: ```suggestion
Scopes are basically a stack of data that is attached to events. When an event is captured, the SDK will merge the data from the active scopes into the event. This allows you to attach data to events that is relevant to the context in which the event was captured.
```
Hope I'm not misunderstanding, but it sounds like data from all active scopes (global, isolation, current) is applied to the event, not just the current scope.

You can access the isolation scope via `Sentry.getIsolationScope()`, but usually you'll just use the `Sentry.setXXX` methods to set data on the currently active isolation scope:

```javascript
Sentry.setTag("my-tag", "my value");
// Is identical to:
Sentry.getIsolationScope().setTag("my-tag", "my value");
```

<PlatformCategorySection supported={["browser"]}>
  In the browser, the isolation scope is never forked, because it is impossible
  to keep track of where an isolation scope would belong to. Because of this, in
  the browser the isolation scope is effectively global.
</PlatformCategorySection>

### Current Scope

> The current scope is the local scope that is currently active. In contrast to the isolation scope, which is usually only forked rarely, e.g. per request, the current scope may be forked more frequently and under the hood. It can be used to store data that should only be applied to specific events. In most cases, you should not access this scope directly, but use `Sentry.withScope` to create a new scope that is only active for a specific part of your code:

Reviewer 2: ```suggestion
The current scope is the local scope that is currently active. Unlike the rarely-forked isolation scope, the current scope may be forked more frequently and under the hood. It can be used to store data that should only be applied to specific events. In most cases, you should not access this scope directly, but use `Sentry.withScope` to create a new scope that is only active for a specific part of your code:
```

```javascript
Sentry.withScope((scope) => {
  // scope is the current scope inside of this callback!
  scope.setTag("my-tag", "my value");
  // this tag will only be applied to events captured inside of this callback
  // the following event will have the tag:
  Sentry.captureException(new Error("my error"));
});
// this event will not have the tag:
Sentry.captureException(new Error("my other error"));
```

> You can access the curent scope via `Sentry.getCurrentScope()`, but usually you should use `Sentry.withScope()` to interact with local scopes instead.

Reviewer 2: ```suggestion
You can access the current scope via `Sentry.getCurrentScope()`, but usually you should use `Sentry.withScope()` to interact with local scopes instead.
```

## How Scope Data is Applied to Events

> Before an event (e.g. an error or transaction) is sent to Sentry, the currently active scopes are applied to it.

Reviewer 2: ```suggestion
Before an event (like an error or transaction) is sent to Sentry, the currently active scopes are applied to it.
```

> The global scope is applied first, followed by the isolation scope, and finally the current scope. This means that any data set on the current scope will take precedence over data set on the isolation or global scope:

Reviewer 2: ```suggestion
The global scope is applied first, followed by the isolation scope, and finally the current scope. This means that any data set on the current scope will take precedence over data set on the isolation and global scopes:
```
